apps/dashboard/src/utils/ip.ts

```diff
@@ -0,0 +1,36 @@
+// IPv4 regex components
+const ipv4Octet = "(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)";
+const ipv4Address = `^${ipv4Octet}\\.${ipv4Octet}\\.${ipv4Octet}\\.${ipv4Octet}$`;
+
+// IPv6 regex components
+const ipv6Segment = "[a-fA-F0-9]{1,4}";
+const ipv6Address = `^(?:
+  (?:${ipv6Segment}:){7}${ipv6Segment}|                                // 1:2:3:4:5:6:7:8
+  (?:${ipv6Segment}:){1,7}:|                                           // 1::                                 1:2:3:4:5:6:7::
+  (?:${ipv6Segment}:){1,6}:${ipv6Segment}|                             // 1::8               1:2:3:4:5:6::8   1:2:3:4:5:6::8
+  (?:${ipv6Segment}:){1,5}(?::${ipv6Segment}){1,2}|                    // 1::7:8             1:2:3:4:5::7:8   1:2:3:4:5::8
+  (?:${ipv6Segment}:){1,4}(?::${ipv6Segment}){1,3}|                    // 1::6:7:8           1:2:3:4::6:7:8   1:2:3:4::8
+  (?:${ipv6Segment}:){1,3}(?::${ipv6Segment}){1,4}|                    // 1::5:6:7:8         1:2:3::5:6:7:8   1:2:3::8
+  (?:${ipv6Segment}:){1,2}(?::${ipv6Segment}){1,5}|                    // 1::4:5:6:7:8       1:2::4:5:6:7:8   1:2::8
+  ${ipv6Segment}:(?:(?::${ipv6Segment}){1,6})|                         // 1::3:4:5:6:7:8     1::3:4:5:6:7:8   1::8
+  :(?:(?::${ipv6Segment}){1,7}|:)|                                     // ::2:3:4:5:6:7:8    ::2:3:4:5:6:7:8  ::8       ::
+  fe80:(?::${ipv6Segment}){0,4}%[0-9a-zA-Z]{1,}|                       // fe80::7:8%eth0     fe80::7:8%1  (link-local IPv6 addresses with zone index)
+  ::(?:ffff(?::0{1,4}){0,1}:){0,1}${ipv4Address}|                      // ::255.255.255.255  ::ffff:255.255.255.255  ::ffff:0:255.255.255.255 (IPv4-mapped IPv6 addresses and IPv4-translated addresses)
+  (?:${ipv6Segment}:){1,4}:${ipv4Address}                              // 2001:db8:3:4::192.0.2.33  64:ff9b::192.0.2.33 (IPv4-Embedded IPv6 Address)
+)$`
+  .replace(/\s*\/\/.*$/gm, "")
+  .replace(/\n/g, "");
```

Comment:
please either find a well tested library that does this or add unit tests for this